### DISCUSSION TITLE
Module federation

### DISCUSSION THREAD

**Discussion by abelta** [stalniy/casl#975] [Q&A]
I'm having a hard time making CASL work with Module Federation in Vite.
Do we know if the library works for sure with that type of architecture and does anyone know of a good example?

**Comment by stalniy** [comment]
Casl modules are all pure. They do not have side effects. So, I don’t see how they can make an issue for federation 

anyway, if you can prepare steps or repository with the issue I’ll take a look

**Comment by abelta** [comment]
I have the project up in three parts with instructions on how to build it and run it.

https://github.com/abelta/vite-federation-host
https://github.com/abelta/vite-federation-remote1
https://github.com/abelta/vite-federation-2

Regarding the architecture: I set host up as a shell for the other two remote apps to be inside. I make sure it has a CASL context and that it uses the same abilities instance for itself and for remote 1 (all configured in vite.config.js). I've managed to make React Router and Zustand work in this setup but not CASL.

It's not like I have a lot of experience with this architecture so I might be doing it just wrong.
